controller/src/beerocks/master/son_management.cpp

```diff
@@ -1720,6 +1720,27 @@ void son_management::handle_bml_message(
         break;
     }
 #endif //BEEROCKS_RDKB
+
+    case beerocks_message::ACTION_BML_TRIGGER_TOPOLOGY_QUERY: {
+
+        auto bml_request = cmdu_rx.addClass<beerocks_message::cACTION_BML_TRIGGER_TOPOLOGY_QUERY>();
+
+        std::string al_mac = network_utils::mac_to_string(bml_request->al_mac());
+
+        auto cmdu_header = cmdu_tx.create(0, ieee1905_1::eMessageType::TOPOLOGY_QUERY_MESSAGE);
+
+        LOG(INFO) << "ACTION_BML_TRIGGER_TOPOLOGY_QUERY al_mac:" << al_mac;
+
+        if (cmdu_header == nullptr) {
+            LOG(ERROR) << "Failed building IEEE1905 TOPOLOGY_QUERY_MESSAGE";
+            return;
+        }
+
+        Socket *agent_sd = database.get_node_socket(al_mac);
```

Comment:
You need to check the return value.

@tomereli BTW it is not clear to me if we should use `!agent_sd` (using implicit conversion to bool) or `== std::nullptr` (using explicit operator ==).

Another comment (not for Pavel): I think the database should be converted to use sMacAddr instead of std::string for the MAC address lookups. Should I make a refactoring issue for that?

Reply:
`!agent_sd` should be used as far as I remember to avoid KW issues.
Regarding refactoring - by all means, go ahead:)